File: Models/Event.cs
﻿using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace group_events_project.Models;

public partial class Event
{
    public int Id { get; set; }

    public string? EventName { get; set; }

    public string? EventLocation { get; set; }

    public double? Price { get; set; }

    public string? EventDescription { get; set; }

    public string? EventImage { get; set; }

    public string? EventCategory { get; set; }

    public string? EventVenue { get; set; }

    public DateTime? EventTime { get; set; }
    //remove infinate loop
    [JsonIgnore]
    public virtual ICollection<Favorite> Favorites { get; set; } = new List<Favorite>();
}

File: ClientApp/src/app/models/event.ts
import { Time } from '@angular/common';

export interface Event {
  id: number;
  eventName: string;
  eventDate: Date;
  eventTime: Date;
  eventLocation: string;
  price: number;
  eventDescription: string;
  eventImage: string;
  eventCategory: string;
  eventVenue: string;

  //favorites: number; //change to account for table join?
  //Can be handled with
}
